src/features/import/components/UploadFile.tsx

```diff
@@ -0,0 +1,177 @@
+import { useDropzone } from 'react-dropzone';
+import {
+  Box,
+  Button,
+  CircularProgress,
+  IconButton,
+  Typography,
+} from '@mui/material';
+import { useCallback, useState } from 'react';
+
+import { ImportedFile } from '../utils/parseFile';
+import messageIds from '../l10n/messagesIds';
+import theme from 'theme';
+import { UploadFileOutlined } from '@mui/icons-material';
+import useImportedFile from '../hooks/useImportedFile';
+import { Msg, useMessages } from 'core/i18n';
+
+const UploadFile = () => {
+  const [uploadedFile, setUploadedFile] = useState<File | null>();
+  const [rawData, setRawData] = useState<ImportedFile>();
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<boolean>(false);
+  const messages = useMessages(messageIds);
+  const { importData } = useImportedFile();
+
+  const onDrop = useCallback((acceptedFiles: File[]): void => {
+    acceptedFiles.map((file: File) => {
+      setLoading(true);
+      setUploadedFile(file);
+
+      importData(file)?.then((res) => {
+        setRawData(res);
+        setLoading(false);
+      });
+    });
+  }, []);
+
+  const { getRootProps, getInputProps, open, isDragActive } = useDropzone({
+    accept: {
+      types: ['.csv', '.xls', '.xlsx'],
+    },
+    maxFiles: 1,
+    noClick: true,
+    noKeyboard: true,
+    onDrop,
+    onDropAccepted: () => setError(false),
+    onDropRejected: () => setError(true),
+    onError: () => {
+      setError(true);
+    },
+  });
+
+  return (
+    <>
+      <Box
+        {...getRootProps({ className: 'dropzone' })}
+        p={12}
+        style={{
+          alignItems: 'center',
+          backgroundColor: error ? '#fdf7f7' : 'transparent',
+          borderColor: error ? theme.palette.error.dark : '#E0E0E0',
+          borderRadius: 4,
+          borderStyle: error ? 'solid' : 'dashed',
+          borderWidth: 2,
+          display: 'flex',
+          flexDirection: 'column',
+          flexWrap: 'wrap',
+          margin: 16,
+          textAlign: 'center',
+        }}
+      >
+        {loading && !rawData && (
+          <Box>
+            <CircularProgress />
+            <Typography>{messages.importDialog.loading()}</Typography>
+          </Box>
+        )}
+        <input className="input-zone" type="file" {...getInputProps()} />
+        <Box className="text-center">
```

Comment:
What's the purpose of these class names (and others around the file)?

Reply:
I just forgot to remove them! Thanks for spotting them!